nixos/modules/services/logging/packetbeat.nix

```diff
@@ -0,0 +1,256 @@
+        '';
+        example = ''
+          output = {
+            elasticsearch = {
+              hosts = [ "localhost:9200" ];
+            };
+          };
+          setup = {
+            template = {
+              settings = {
+                index.number_of_shards =  1;
+              };
+            };
+            kibana = {
+              host =  "localhost:5601";
+            };
+          };
+          processors =  [
+            '''
+              if.contains.tags: forwarded
+              then:
+                - drop_fields:
+                  fields: [host]
+              else:
+                - add_host_metadata: ~
+            '''
+            "add_cloud_metadata: ~"
+            "add_docker_metadata: ~"
+          ];
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.packetbeat.settings = {
+      packetbeat = {
+        interfaces.device = mkDefault "any";
+        flows = {
+          timeout = mkDefault "30s";
+          period = mkDefault "10s";
+        };
+        protocols = mkDefault (attrsToList {
+          icmp = {
+            enabled = true;
+          };
+          amqp = {
+            ports = [ 5672 ];
+          };
+          cassandra = {
+            ports = [ 9042 ];
+          };
+          dhcpv4 = {
+            ports = [ 67 68 ];
+          };
+          dns = {
+            ports = [ 53 ];
+          };
+          http = {
+            ports = [ 80 8080 8000 5000 8002 ];
+          };
+          memcache = {
+            ports = [ 11211 ];
+          };
+          mysql = {
+            ports = [ 3306 3307 ];
+          };
+          pgsql = {
+            ports = [ 5432 ];
+          };
+          redis = {
+            ports = [ 6379 ];
+          };
+          thrift = {
+            ports = [ 9090 ];
+          };
+          mongodb = {
+            ports = [ 27017 ];
+          };
+          nfs = {
+            ports = [ 2049 ];
+          };
+          tls = {
+            ports = [ 443 993 995 5223 8443 8883 9243 ];
+          };
+        });
+      };
+      output = {
+        elasticsearch = {
+          hosts = mkDefault [ "localhost:9200" ];
+        };
+      };
+      setup = {
+        template = {
+          settings = {
+            index.number_of_shards = mkDefault 1;
+          };
+        };
+        kibana = {
+          host = mkDefault "localhost:5601";
+        };
+      };
+      processors = mkDefault [
+        ''
+          if.contains.tags: forwarded
+          then:
+            - drop_fields:
+              fields: [host]
+          else:
+            - add_host_metadata: ~
+        ''
+        "add_cloud_metadata: ~"
+        "add_docker_metadata: ~"
+      ];
+    };
+
+    systemd.services.packetbeat = {
+      description = "Packetbeat log shipper";
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = mkMerge [
+        {
+          ExecStart = ''
+            ${cfg.package}/bin/packetbeat \
```

Comment:
You mentioned previously that the official systemd unit runs as root in https://github.com/NixOS/nixpkgs/pull/94862#discussion_r467458109, but I believe it's still worth trying to make it work without `root`, because it's really not very good to use `root`. Ideally `DynamicUser` would be nice.

Reply:
Adding DynamicUser = "yes" and CAP_NET_RAW and CAP_NET_ADMIN in AmbientCapabilites solved this.